php-checks/src/main/java/org/sonar/php/checks/CryptographicKeySizeCheck.java

```diff
@@ -0,0 +1,109 @@
+import org.sonar.plugins.php.api.tree.expression.ArrayInitializerTree;
+import org.sonar.plugins.php.api.tree.expression.ArrayPairTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+import static org.sonar.php.checks.utils.CheckUtils.trimQuotes;
+
+@Rule(key = CryptographicKeySizeCheck.KEY)
+public class CryptographicKeySizeCheck extends PHPVisitorCheck {
+
+  public static final String KEY = "S4426";
+  private static final int MIN_KEY_LENGTH = 2048;
+  private static final String MESSAGE = "Use a key length of at least " + MIN_KEY_LENGTH + " bits";
+
+  private AssignmentExpressionVisitor assignmentExpressionVisitor;
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    this.assignmentExpressionVisitor = new AssignmentExpressionVisitor(context().symbolTable());
+    tree.accept(assignmentExpressionVisitor);
+    super.visitCompilationUnit(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree functionCall) {
+    if ("openssl_pkey_new".equals(CheckUtils.getFunctionName(functionCall))) {
+      SeparatedList<ExpressionTree> arguments = functionCall.arguments();
+      if (arguments.size() == 1) {
+        ExpressionTree configArgument = arguments.get(0);
+        getKeySize(configArgument)
+          .filter(this::lessThanMinKeyLength)
+          .ifPresent(keySize -> context().newIssue(this, keySize, MESSAGE));
+      }
+    }
+    super.visitFunctionCall(functionCall);
+  }
+
+  private boolean lessThanMinKeyLength(ExpressionTree keySize) {
+    if (keySize.is(Kind.NUMERIC_LITERAL)) {
+      LiteralTree literal = (LiteralTree) keySize;
+      int size = Integer.parseInt(literal.value());
+      return size < MIN_KEY_LENGTH;
+    } else if (keySize.is(Kind.VARIABLE_IDENTIFIER)) {
+      return assignmentExpressionVisitor.getAssignmentValue(keySize)
+        .map(this::lessThanMinKeyLength)
+        .orElse(false);
+    }
+    return false;
+  }
+
+  private Optional<ExpressionTree> getKeySize(ExpressionTree config) {
+    if (config.is(Kind.ARRAY_INITIALIZER_FUNCTION, Kind.ARRAY_INITIALIZER_BRACKET) && isRSA((ArrayInitializerTree) config)) {
+      return ((ArrayInitializerTree) config).arrayPairs().stream()
+        .filter(pair -> pair.key() != null && pair.key().is(Kind.REGULAR_STRING_LITERAL) && "private_key_bits".equals(trimQuotes((LiteralTree) pair.key())))
+        .map(ArrayPairTree::value)
+        .findFirst();
+    }
+    return assignmentExpressionVisitor
+      .getAssignmentValue(config)
+      .flatMap(this::getKeySize);
+  }
+
+  private boolean isRSA(ArrayInitializerTree config) {
+    return config.arrayPairs().stream()
+      .filter(pair -> {
+        if (!"private_key_type".equals(trimQuotes((LiteralTree) pair.key()))) {
```

Comment:
This throws a `ClassCastException` when `pair.key()` is not a literal. We should have a test for that.
Moreover, `pair.key()` may return null.